### PR TITLE
Rename Protomatter to RGBMatrix

### PR DESCRIPTION
**Pull request by jepler**
~~In the interest of time, this only renames the user facing parts, a cleanup PR to rename internally will come later on.~~
This renames the user-facing parts as well as the identifiers that are internal to CircuitPython.

Closes: #2776

### CODE REVIEW

**Comment by jepler** [comment]
This also  has the changes from #2774

**Comment by jepler** [comment]
User facing names:
```
>>> import rgbmatrix
>>> rgbmatrix
<module 'rgbmatrix'>
>>> rgbmatrix.RGBMatrix
<class 'RGBMatrix'>
```

Constructing and using:
```
matrix = rgbmatrix.RGBMatrix(
    width=64, height=32, bit_depth=3,
    rgb_pins=[board.D6, board.D5, board.D9, board.D11, board.D10, board.D12],
    addr_pins=[board.A5, board.A4, board.A3, board.A2],
    clock_pin=board.D13, latch_pin=board.D0, output_enable_pin=board.D1)
display = framebufferio.FramebufferDisplay(matrix, auto_refresh=False)
```

File: shared-bindings/rgbmatrix/RGBMatrix.c [hunk 1]
```diff
@@ -133,11 +133,11 @@ STATIC void preflight_pins_or_throw(uint8_t clock_pin, uint8_t *rgb_pins, uint8_
     }
 }
 
-//| :class:`~_protomatter.Protomatter` displays an in-memory framebuffer to an LED matrix.
+//| :class:`~rgbmatrix.RGBMatrix` displays an in-memory framebuffer to an LED matrix.
 //|
-//| .. class:: Protomatter(width, bit_depth, rgb_pins, addr_pins, clock_pin, latch_pin, output_enable_pin, *, doublebuffer=True, framebuffer=None)
+//| .. class:: RGBMatrix(width, bit_depth, rgb_pins, addr_pins, clock_pin, latch_pin, output_enable_pin, *, doublebuffer=True, framebuffer=None)
```
Comment (tannewt) [inline shared-bindings/rgbmatrix/RGBMatrix.c:138]: I thought we added height here. Did it get lost in the PR shuffle?

Comment (jepler) [reply shared-bindings/rgbmatrix/RGBMatrix.c]: Docs got out of date.  Fixed now?

**Comment by jepler** [comment]
At this point, Actions reports '80 successful, 43 queued, and 1 in progress checks'.  build-arm (circuitplayground_bluefruit) started 19h 17m 4s ago and has been running the step 'build'  for 19h 15m 35s.  The log for the build step can't be viewed.

I'm going to cancel and re-run, and if that doesn't break it loose, I'll push a nonessential change.  Hopefully one of those will get this on the road again.

**Comment by jepler** [comment]
Rebased and "make translate"d again.

**Comment by jepler** [comment]
In my own fork's actions, 9bfe6b7 (framebufferio: update copyright information) succeeded instead of hanging: https://github.com/jepler/circuitpython/actions/runs/80760861 -- however, for circuitpython it was hung and I cancelled it.

**Comment by jepler** [comment]
@tannewt Again, this PR build hung.    I've made another non-essential change.

**Comment by jepler** [comment]
@tannewt actions finally finished this time!

**Pull request merged**